### PR TITLE
Teach page form, validate, fetch and error handling

### PR DESCRIPTION
**Pull request by SarKurd**
- Connected the form to the back-end
- Validated the user input
- Handled the errors
- Fixed the focusing issue

### CODE REVIEW

File: pages/teach/index.vue [hunk 1]
```diff
@@ -112,11 +126,37 @@ export default {
...
-    components: {},
     methods: {
+        async submitForm(e) {
+            const fields = {};
+            Object.values(e.target.elements).forEach(
+                input => (fields[input.name] = input.value)
+            );
+
+            try {
+                await axios.post(process.env.lambdaUrl + "teach", fields);
```
Comment (isaachinman) [inline pages/teach/index.vue]: Would be nice to see a template literal here.

File: pages/teach/index.vue [hunk 2]
```diff
@@ -112,11 +126,37 @@ export default {
...
+                this.$el
+                    .querySelectorAll(".inputContainer")
+                    .forEach((i, index) => {
+                        if (index !== 2) {
+                            i.childNodes[1].value = "";
+                            i.classList.remove("active");
+                        }
+                    });
+            } catch (error) {
+                console.log(error.message);
```
Comment (isaachinman) [inline pages/teach/index.vue]: Please talk to @gisaac85 about error handling. He is currently implementing the same functionality for the apply form. The API should return useful validation messages, so we want to make those visible to the user via the UI.

File: pages/teach/index.vue (hunk 1 above)
Comment (SarKurd) [reply pages/teach/index.vue]: Will change it

File: pages/teach/index.vue (hunk 2 above)
Comment (SarKurd) [reply pages/teach/index.vue]: it's already done and there, i just have to make a small change, i thought showing `something went wrong` is enough for the user

File: pages/teach/index.vue [hunk 3]
```diff
@@ -146,7 +146,9 @@ export default {
                         }
                     });
             } catch (error) {
-                console.log(error.message);
+                if (error.response.data.message) {
+                    this.failedMessage = error.response.data.message;
```
Comment (isaachinman) [inline pages/teach/index.vue]: Isn't the error response an array? Can you please talk with @Aya-Alabrash and make sure that we have one single way of error handling in these forms?

Comment (isaachinman) [reply pages/teach/index.vue]: The same feedback applies to the contact page as well. Just want to make sure everyone is doing the same kind of validation so it feels consistent.

Comment (SarKurd) [reply pages/teach/index.vue]: When i wrote the back-end validation i returned whole response with the unnecessary data, i believe it was wrong so i changed it to a single string for Teach and Contact, haven't touched Apply yet, i believe it's easier and more clear to handle it in the front

**Comment by SarKurd** [comment]
Hey Travis

**Pull request merged**